### DISCUSSION TITLE
DMatrix row_sum() cast to different type

### DISCUSSION THREAD

**Discussion by jon-ton** [dimforge/nalgebra#1183] [General]
Hello and thanks for all the good work on this crate! 

Is there a preferred way to do a row_sum() cast to a different type? For example, if I have a 28x28 DMatrix of type u8, my row_sum() will overflow. I've been manually iterating over the matrix to get the row_sum() cast as a u32, but figured I'd ask if there was an easier way with the crate and some type of .into() syntax.

**Comment by sebcrozet** [comment]
Hi! You could do something like this:

```rust
self.compress_rows(|col| col.iter().map(|e| *e as u32).sum())
```

It’s basically the current implementation of `row_sum`, but with a cast inside before summing the column’s elements.

**Comment by jon-ton** [comment]
Thanks for the help.
